### DISCUSSION TITLE
Set a field as immutable

### DISCUSSION THREAD

**Discussion by CovertCode** [pocketbase/pocketbase#4075] [Q&A] [answered]
I am using pocketbase for a small scale game. I want to set a field as immutable. I know this isn't a feature of sqlite but I guess it is possible to do it using hook: before-record-update.
If I run a check using **`onRecordBeforeUpdateRequest`** hook to check whether the field which I want to be only writeable once is present in update request and based on this check I will allow record update. What could be downsides of running this hook for each update?

Note: I am test this once I reach home. So I thought to ask the community for some suggestions.

**Comment by ganigeorgiev** [comment, marked as answer]
It depends on your requirements but there shouldn't be any significant performance impact when using the hooks.

An alternative to the hooks could be to use the API rules. For example, if your field is a `text` field, one such Update API rule could be:

```js
// the existing value is empty (remove if users are not allowed to update empty values)
yourField = "" ||
// the submitted request doesn't have yourField
@request.data.yourField:isset = false ||
// the submitted request has yourField but it matches with the existing value
@request.data.yourField = yourField
```
_(note that failed API rules returns a generic error)_

**Comment by CovertCode** [comment]
If anyone wants to implement this via hooks then you can get some help from here: https://github.com/pocketbase/pocketbase/discussions/4077#discussioncomment-8011216

and to prevent non-admin from updating this you can use API filter: `@request.auth.id != "" && @request.data.yourField:isset = false`
